### ISSUE TITLE
error when using traverseOnVFG to collect all LLVM Values

### ISSUE THREAD

**Issue by yikuoyansui** [SVF-tools/SVF-example#30]
Hi,

I tried to use traverseOnVFG to collect all LLVM values with regard to one specific VFG node.
I kept getting this error "svf-ex: /home/travis/build/SVF-tools/SVF/lib/Graphs/VFG.cpp:868: const SVF::PAGNode* SVF::VFG::getLHSTopLevPtr(const SVF::VFGNode*) const: Assertion `false && "unexpected node kind!"' failed.", I am wondering whether you could provide any hint on it.

Thanks

**Comment by yuleisui** [comment]
You need to provide more information about the SVFGNode and its PAGNode information so that I could understand which SVFGNode you are referring to.

**Comment by yikuoyansui** [comment]
Thanks for your quick reply. The program is too large. It is hard to give you specific information about the SVFGNode and its PAGNode without context. The following three statements show how I fetch the llvm value and feed it into traverseOnVFG. 

PAGNode* pagNode = pag->getPAGNode(5394);
const Value* v =  pagNode->getValue();
traverseOnVFG(svfg, v);

I use the node ID (5394) which can be obtained from pag.dot to get the corresponding PAGNode. The following snippet is information about this node.

Node0x55dd51a85c00 [shape=record,shape=circle,label="{[base64_decode] 5394:}"];
	Node0x55dd51a85c00 -> Node0x55dd51a7f6c0[color=blue];

I use several types of nodes as the input and all of them show the same error.

If the above description is hard to follow, could you give me some high-level hints about how to properly use traverseOnVFG?

Thanks

**Comment by yuleisui** [comment]
The addresses would tell the problem. Can you send me your bc?

**Comment by yikuoyansui** [comment]
Btw, I am wondering why we need to use getLHSTopLevPtr considering we have already traversed VFG and get the visited list before.

**Comment by yuleisui** [comment]
You can also use `node->toString()` to get the results for your purpose. No need to use `getLHSTopLevPtr`.

**Comment by yikuoyansui** [comment]
I am confused that what the purpose of getLHSTopLevPtr is. Is it to get the corresponding PAGNode for VFGNode and PAGNode are the exact LLVM values (pointers), right?

**Comment by yuleisui** [comment]
In brief, yes. See the fixed patch for SVF here: https://github.com/SVF-tools/SVF/commit/75e3b7c2c34045700cbf619a64991d99c3398212

and the fix for SVF-example here:
https://github.com/SVF-tools/SVF-example/commit/1b5edfbe9fdfc6461ea1abba11e5fc668bba2297

**Comment by yikuoyansui** [comment]
Thanks for your updates. I updated all the patches and fix, while the same error still happens. Has the error been fixed at your side? I am using the same node 5394 and the same bc.

svf-ex: /home/travis/build/SVF-tools/SVF/lib/Graphs/VFG.cpp:874: const SVF::PAGNode* SVF::VFG::getLHSTopLevPtr(const SVF::VFGNode*) const: Assertion `false && "unexpected node kind!"' failed.

Thanks

**Comment by yuleisui** [comment]
You will need to update svf-lib `npm i --silent svf-lib --prefix ${HOME}`

**Comment by yikuoyansui** [comment]
<img width="657" alt="Screen Shot 2020-11-16 at 11 33 49 PM" src="https://user-images.githubusercontent.com/31490895/99347102-31117380-2864-11eb-8520-1f3faba2d983.png">
I do execute the above command. I did one more check on the update of libSvf.a and I found the time remains the old time which is not up-to-date. I do not know why.

**Comment by yikuoyansui** [comment]
Sorry to disturb you again. I did execute the above command. I checked the modification time of libSvf.a under Release-build folder and the time is no up-to-date. I am not sure whether the problem is caused by that. Thanks.

**Comment by yuleisui** [comment]
@JasonZhongZexin could you take a look at the npm configuration?

**Comment by JasonZhongZexin** [comment]
@yikuoyansui I have confirmed the new update has been published to NPM. The up-to-date version of svf-lib should be v1.0.243. Could you please provide the below information, so that I could investigate the issue. 
1.  What‘s the version of your OS?
2. What’s the version of the npm and nodejs that you are using?
3. What’s the version of the installed svf-lib?
4. What’s the output that display on your terminal after running the svf-lib update command?
